akka-http2-support/src/main/scala/akka/http/impl/engine/http2/Http2Blueprint.scala

```diff
@@ -79,10 +79,7 @@ object Http2Blueprint {
     // This is master header parser, every other usage should do .createShallowCopy()
     // HttpHeaderParser is not thread safe and should not be called concurrently,
     // the internal trie, however, has built-in protection and will do copy-on-write
-    val masterHttpHeaderParser = HttpHeaderParser(parserSettings, log) { info ⇒
-      if (parserSettings.illegalHeaderWarnings)
-        logParsingError(info withSummaryPrepended "Illegal request header", log, parserSettings.errorLoggingVerbosity)
-    }
+    val masterHttpHeaderParser = HttpHeaderParser(parserSettings, log)
```

Comment:
I could've sworn I changed that "illegal response header" message to "illegal header", but there you go. Will fix in a second. 

As far as differentiating between response and request header warnings, it's possible to do so by providing a custom message as an argument to log in case of illegal header, but I figured that introducing a parameter just for that was a little overkill as this would mostly be obvious from the origin of the warning.